tests/validation/tests/v3_api/test_nodepool.py

```diff
@@ -0,0 +1,73 @@
+import ast
+import os
+import pytest
+from .common import get_user_client
+from .common import get_user_client_and_cluster
+from lib.aws import AmazonWebServices
+from .test_rke_cluster_provisioning import random_node_name
+from .test_rke_cluster_provisioning import node_template_ec2
+from .test_rke_cluster_provisioning import create_and_validate_cluster
+from .test_rke_cluster_provisioning import rke_config
+from .test_ebs_volume_backed_instance import get_aws_nodes_from_nodepools
+DRAIN_BEFORE_DELETE = ast.literal_eval(
+    os.environ.get('RANCHER_DRAIN_BEFORE_DELETE', "False")
+    )
+
+def test_nodepool_delete():
+    node_template = node_template_ec2()
+    client = get_user_client()
+    nodes = []
+    node = get_node_pool(node_template, "controlPlane", 1)
+    nodes.append(node)
+    node = get_node_pool(node_template, "etcd", 1)
+    nodes.append(node)
+    node = get_node_pool(node_template, "worker", 3)
+    nodes.append(node)
+    node = get_node_pool(node_template, "worker", 2)
+    nodes.append(node)
+    cluster, node_pools = create_and_validate_cluster(
+        client, nodes, rke_config
+        )
+    #delete a worker nodepool
+    for nodepool in node_pools:
+        if nodepool["worker"]:
+            nodes = client.list_node(nodePoolId=nodepool.id)
+            aws_nodes = []
+            for node in nodes:
+                print("node.externalIpAddress: ", node.externalIpAddress)
+                filters = [{'Name': 'ip-address','Values': [node.externalIpAddress]}]
+                node_details = AmazonWebServices().get_nodes(filters)
```

Comment:
`node_details` seem redundant here, this can be directly assigned to `aws_nodes` or if really needed `aws_nodes.extend(node_details)`.

Then we can do `assert len(aws_nodes) > 0`... etc.

Reply:
done